### PR TITLE
editoast: rename `Unauthenticated` to `Unauthorized`, `Unauthorized` to `Forbidden`

### PR DESCRIPTION
**Pull request by hamz2a**
This PR updates the error messages to better align with HTTP status codes. Specifically, it renames the `Unauthenticated` error to `Unauthorized` and the `Unauthorized` error to `Forbidden`. This change improves clarity and consistency in our error handling.

### CODE REVIEW

File: editoast/src/views/mod.rs [hunk 1]
```diff
@@ -241,12 +241,12 @@ async fn authentication_middleware(
...
 #[editoast_error(base_id = "authz")]
 pub enum AuthorizationError {
-    #[error("Unauthenticated")]
-    #[editoast_error(status = 401)]
-    Unauthenticated,
     #[error("Unauthorized")]
-    #[editoast_error(status = 403)]
+    #[editoast_error(status = 401)]
     Unauthorized,
+    #[error("Forbidden")]
```
Comment (leovalais) [inline editoast/src/views/mod.rs]:
```suggestion
    #[error("Forbidden — user has insufficient privileges")]
```

File: editoast/src/views/mod.rs [hunk 2]
```diff
@@ -241,12 +241,12 @@ async fn authentication_middleware(
 #[derive(Debug, Error, EditoastError)]
 #[editoast_error(base_id = "authz")]
 pub enum AuthorizationError {
-    #[error("Unauthenticated")]
-    #[editoast_error(status = 401)]
-    Unauthenticated,
     #[error("Unauthorized")]
```
Comment (leovalais) [inline editoast/src/views/mod.rs]:
```suggestion
    #[error("Unauthorized — user must be authenticated")]
```

**Review by leovalais — APPROVED** [review]

**Comment by leovalais** [comment]
Your commit message is too long: it must be <= 72 characters

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Pull request merged**